### ISSUE TITLE
Standard names: Virtual potential temperature

### ISSUE THREAD

**Issue by SimonPeatman** [cf-convention/discuss#194]
Most of the variants of air temperature have a corresponding standard name for potential temperature (theta), for example:
- ```air_temperature``` and ```air_potential_temperature```
- ```air_equivalent_temperature``` and ```air_equivalent_potential_temperature```
- ```air_pseudo_equivalent_temperature``` and ```air_pseudo_equivalent_potential_temperature```
- ```wet_bulb_temperature``` and ```wet_bulb_potential_temperature```

But although there is ```virtual_temperature``` there is no ```virtual_potential_temperature```.

**Proposer's name** Simon Peatman
**Date** 9 December 2022
**- Term** ```virtual_potential_temperature```
**- Description** The virtual potential temperature of air is the potential temperature at which the dry air constituent of a parcel of moist air would have the same density as the moist air at the same pressure.
**- Units** K

**Comment by feggleton** [comment]
Hi @SimonPeatman, thank you for your proposal and apologies for the delay in getting to this issue. This is a very good point you have raised and a reasonable request. Standard Names are only added on request when they are needed by the community so there clearly hasn't been a need for this term until now. This looks to follow the definition of the existing name and so I am happy with this. I can't see any phrases which need to be added. @japamment can you add to the editor and confirm happy with this simple addition. 

If there are no further comments then this can be accepted in 7 days. Thanks

**Comment by JonathanGregory** [comment]
Thanks for the proposal. I agree it makes sense to have this.

I suppose that virtual temperature is a concept that applies only in the atmosphere, but nonetheless should the existing name be `air_virtual_temperature` and the new one `air_virtual_potential_temperature`, to be consistent with the other kinds of air temperature, which say `air`?

Jonathan

**Comment by efisher008** [comment]
Hi @SimonPeatman, @JonathanGregory , @feggleton,

Was there further discussion offline/at the CF workshop on this issue that isn't captured here? Or further comments to add regarding @JonathanGregory's suggestion (`air_virtual_potential_temperature` as the new name and changing the existing name to `air_virtual_temperature`)?

Best regards,
Ellie

**Comment by efisher008** [comment]
Hi all, 

@japamment @feggleton perhaps one of you would like to comment on this issue?

To summarise, @SimonPeatman has proposed the addition of a new standard name `virtual_potential_temperature` to complement the existing `virtual_temperature`.

@JonathanGregory has suggested that the term _'air'_ should be added to both the new name and the existing name, i.e. aliasing `virtual_temperature` and renaming it `air_virtual_temperature`, then Simon's new standard name would be `air_virtual_potential_temperature`, to be consistent with other standard names referring to air temperature (outlined in the original post). I would tend to agree with this as it makes everything clearer re. the medium the quantity is describing.

Best regards,
Ellie

**Comment by taylor13** [comment]
To my ear ``virtual_air_temperature`` and ``virtual_air_potential_temperature`` sound better. Or perhaps,
 ``virtual_temperature_of_air`` and ``virtual_potential_temperature_of_air``.   (or even replace "of" with "in").  What do others think?

**Comment by JonathanGregory** [comment]
I agree with Karl @taylor13 that `virtual_temperature_of_air` and `virtual_potential_temperature_of_air` would be better that what I suggested. I think `of` is right on the basis of other names e.g. `specific_dry_energy_of_air`. `virtual_air_potential_temperature` sounds like the air might be virtual. :smile:

**Comment by efisher008** [comment]
Hello @taylor13 and @JonathanGregory,

Thank you for your comments. I've done a quick literature search and "_virtual temperature (of air)_" appears quite often in that construction (there are also examples of "_air virtual temperature_" and "_virtual air temperature_" being used).
I can see the benefit of having `virtual_temperature` as an unbroken searchable phrase in the standard name as this seems like it would result in easier findability in the standard names table. The recommendation at the moment is therefore to:

1. Existing standard name (`virtual_temperature`) be renamed to: `virtual_temperature_of_air`.

2. New standard name to be added: `virtual_potential_temperature_of_air`.

Does that make sense?

Best regards,
Ellie

**Comment by JonathanGregory** [comment]
That makes sense to me. Thanks, Ellie.

**Comment by efisher008** [comment]
Dear @JonathanGregory and @SimonPeatman,

While talking to @japamment about this issue earlier this week, we were in two minds about which construction is actually more suitable in this case: `temperature_of_air` or `air_temperature`. Both do appear in the standard names vocabulary, but according to the CF conventions the 'proper' construction should be `in_air`. I'll let Alison give her opinion before going ahead with these. 

Best,
Ellie

**Comment by JonathanGregory** [comment]
Dear Ellie @efisher008 

I think it depends on whether it's a property that is defined `in_air` as a medium, or a property `of_air` as a substance. It's true there are a lot more `in_air` than `of_air` existing names. There are, however, some examples of the latter pattern, including `specific_dry_energy_of_air`, `freezing_temperature_of_sea_water` and `compressive_strength_of_sea_ice`. I think `of_air` is better for a temperature. I would say that the existing name `air_temperature` means "temperature _of_ air", not "temperature _in_ air". Its description says "Air temperature is the bulk temperature **of** the air."

Best wishes

Jonathan

**Comment by taylor13** [comment]
I agree with Jonathan's reasoning for temperature, that "of air" is preferred over "in air".

**Comment by efisher008** [comment]
Hi @JonathanGregory and @taylor13,

In that case, and given there haven't been any more comments on this issue, I think the conclusion is as in this post:

1. Existing standard name (`virtual_temperature`) be renamed to: `virtual_temperature_of_air`.

2. New standard name to be added: `virtual_potential_temperature_of_air`.

@japamment would you like to comment on this issue? Otherwise, if there aren't any further comments in the next 7 days, I think we can accept these changes. 

Best wishes,
Ellie

**Comment by japamment** [comment]
Hi @efisher008, @JonathanGregory, @taylor13,

If we are choosing between "in_air" and "of_air" I would certainly agree that "of_air" is more appropriate because the temperature is a property of the air itself.

However, my thinking is that we should go with Jonathan's first suggestion of `air_virtual_temperature` and  `air_virtual_potential_temperature` because this construction is more consistent with other existing temperature names (`air_temperature`, `air_temperature_anomaly`, `air_equivalent_temperature`, `air_pseudo_equivalent_temperature`, and many more). I don't think we should adopt a different construction just for virtual temperature. Also, virtual temperature isn't something that can only be calculated in air - we do have standard names for sea_water_potential_temperature and using the air_potential_temperature syntax would be consistent with these.

@SimonPeatman proposed a description for virtual_potential_temperature: 
>  The virtual potential temperature of air is the potential temperature at which the dry air constituent of a parcel of moist air would have the same density as the moist air at the same pressure.

To this I think we should add wording similar to the existing potential_temperature description: "Potential temperature is the temperature a parcel of air would have if moved dry adiabatically to a standard pressure, typically representative of mean sea level pressure. To specify the standard pressure to which the quantity applies, provide a scalar coordinate variable with standard name reference_pressure." Perhaps others can comment on whether this is correct, but I assume the virtual potential temperature has to be calculated relative to some reference pressure in the same way as potential temperature?

@SimonPeatman mentions `wet_bulb_temperature` and `wet_bulb_potential_temperature` in the original proposal. If we are updating the virtual temperature names by adding a medium then I suggest the wet bulb names should be similarly updated to `air_wet_bulb_temperature` and `air_wet_bulb_potential_temperature`. I don't think wet_bulb temperature could ever make sense in a medium other than air, but again it's about being consistent with the names.

In the process of writing this I noticed that wet_bulb_temperature is one of the small number of standard names that has no description text (it is one of the oldest names in the table). I propose giving it the following description: "Wet bulb temperature  is read from the wet-bulb thermometer in a psychrometer. The temperature measured by the wet-bulb thermometer is generally lower (due to evaporation of water from the wet bulb) than that measured by the dry bulb. The difference in the temperatures is a measure of the humidity of the air; the lower the ambient humidity, the greater the rate of evaporation and, consequently, the greater the depression of the wet-bulb temperature. Reference: https://glossary.ametsoc.org/wiki/Psychrometer."

Best wishes,
Alison

**Comment by efisher008** [comment]
Hi @JonathanGregory, @taylor13, @SimonPeatman,

Would you like to comment on @japamment's [post](https://github.com/cf-convention/discuss/issues/194#issuecomment-1911756134)? Do you agree with using the `air_potential_temperature` syntax for the two names we have been discussing, changing the two `wet_bulb` names to this syntax and/or adding the suggested description to wet bulb temperature? 

Best wishes,
Ellie

**Comment by JonathanGregory** [comment]
Dear Karl @taylor13 

My rationale for suggesting `air` is by analogy to `air_potential_temperature`. Potential temperature is also defined for `sea_water` (as Alison @japamment remarks). It seems inconsistent to remove the `air` when `virtual` is added. Also, although you're right (I think) that "virtual temperature" is defined only for air, the fact that both you and I feel we need to say "(I think)" means this is not perfectly common knowledge we can expect everyone to be sure of. Since standard names are intended to be self-explanatory to all geoscientists, giving them the information that this is a property of air seems reasonable. On the other hand, I expect that people who aren't familiar with the relevant temperatures will nonetheless guess that "dew" and "wet bulb" refer to the atmosphere, not the ocean. There is no dew in the ocean, and a thermometer bulb in the ocean would always be wet.

It's OK with me if we keep `air` as a prefix in these new names, following `air_equivalent_temperature`, which Alison has reminded us about. That one also doesn't occur with `sea_water`.

Best wishes

Jonathan

**Comment by Armin-RS** [comment]
From a physicists view, you could as well specify a dew point temperature in any gas mixed with water vapor, there is no restriction to the specific gas mixture called air.

**Comment by taylor13** [comment]
Good point.  Thanks for raising it.   If we want to extend CF beyond "Climate and Forecast" data, we would probably want to include "air" in the name.  Not sure this is necessary though.

**Comment by JonathanGregory** [comment]
Yes, thanks, I agree. We can add `in_air` later via an alias if there is a use-case that requires the distinction.

**Comment by JonathanGregory** [comment]
I agree with your summary, thanks Ellie @efisher008

**Comment by taylor13** [comment]
this is acceptable to me.  Thanks.

**Comment by japamment** [comment]
Hi @efisher008,

Many thanks for your summary. I don't want to hold this issue up further so I'm willing to go with the majority view as regards the names.

I think in general that we should consider how accessible/understandable our standard names are to communities outside CF given that there is interest in using them more broadly as an environmental science vocabulary. However, I'm happy to take that up in another issue!

Regarding virtual_potential_temperature and wet_bulb_temperature I made a couple of suggestions regarding the descriptions - are there any objections to adding the text (copied again below)?
> 
> @SimonPeatman proposed a description for virtual_potential_temperature:
> 
> > The virtual potential temperature of air is the potential temperature at which the dry air constituent of a parcel of moist air would have the same density as the moist air at the same pressure.
> 
> To this I think we should add wording similar to the existing potential_temperature description: "Potential temperature is the temperature a parcel of air would have if moved dry adiabatically to a standard pressure, typically representative of mean sea level pressure. To specify the standard pressure to which the quantity applies, provide a scalar coordinate variable with standard name reference_pressure." 

> 
> In the process of writing this I noticed that wet_bulb_temperature is one of the small number of standard names that has no description text (it is one of the oldest names in the table). I propose giving it the following description: "Wet bulb temperature is read from the wet-bulb thermometer in a psychrometer. The temperature measured by the wet-bulb thermometer is generally lower (due to evaporation of water from the wet bulb) than that measured by the dry bulb. The difference in the temperatures is a measure of the humidity of the air; the lower the ambient humidity, the greater the rate of evaporation and, consequently, the greater the depression of the wet-bulb temperature. Reference: https://glossary.ametsoc.org/wiki/Psychrometer."

N.B. I see that wet_bulb_temperature is already in the CEDA editor linked to the discussion in  [#270](https://github.com/cf-convention/discuss/issues/270) but I don't think the suggestion above clashes in any way with the text about units.
 
Best wishes,
Alison

**Comment by taylor13** [comment]
For virtual potential temperature, I think the description proposed by @SimonPeatman might be clearer (in my muddled brain) if it read:  "In theory the virtual potential temperature can be obtained by removing the moisture from an air parcel at constant pressure and temperature, decreasing its temperature until the parcel's density is the same as the density of the original parcel, and then bringing that parcel adiabatically and reversibly to some reference pressure, typically 100 kPa." 

I'm actually not 100% sure this is right though.  You might need to first bring the air to the reference pressure adiabatically, then remove the moisture and decrease the temperature to recover the density the parcel had before removal of moisture.  Does anyone know?  Maybe it doesn't matter; you might get the same answer either way.

Perhaps this would be an option:  "The virtual potential temperature of a moist air parcel is theoretically obtained and is the potential temperature of a similar dry air parcel that has been cooled so that its density is the same as in the moist air parcel."

Hopefully, someone can find a better description than any of these.  The main part of the original description that confused me was when it said "temperature at which the dry air constituent of a parcel".  This made me think that there was still a water vapor constituent present (possibly at a different temperature).

**Comment by taylor13** [comment]
Regarding wet_bulb_temperature, I think we mean specifically "isobaric wet-bulb temperature".  The AMS Glossary of Meteorology gives a definition that is more theoretically based (rather than empirically based): "Isobaric wet-bulb temperature: the temperature an air parcel would have if cooled adiabatically to saturation at constant pressure by evaporation of water into it, all latent heat being supplied by the parcel."
